SPECS/systemd/systemd.spec

```diff
@@ -1,62 +1,342 @@
-%define majver %(echo %{version} | cut -d. -f 1)
-Summary:        Systemd
+
+# Temporary until azure-release-common is updated to carry macros.dist that defines %azure
+%global azure 3
+
+# We ship a .pc file but don't want to have a dep on pkg-config. We
+# strip the automatically generated dep here and instead co-own the
+# directory.
+%global __requires_exclude pkg-config
+
+%global pkgdir %{_prefix}/lib/systemd
+%global system_unit_dir %{pkgdir}/system
+%global user_unit_dir %{pkgdir}/user
+
+%if 0%{?__isa_bits} == 64
+%global elf_bits (64bit)
+%global elf_suffix ()%{elf_bits}
+%endif
+
+%bcond bzip2  1
+%bcond gnutls 0
+%bcond lz4    1
+%bcond xz     1
+%bcond zlib   1
+%bcond zstd   1
+
+# Bootstrap may be needed to break circular dependencies with cryptsetup,
+# e.g. when re-building cryptsetup on a json-c SONAME-bump.
+%bcond bootstrap 0
+%bcond tests     1
+%bcond lto       1
+
+# Support for quick builds with rpmbuild --build-in-place.
+# See README.build-in-place.
+%bcond inplace 0
+
 Name:           systemd
-Version:        254.5
-Release:        1%{?dist}
-License:        LGPLv2+ AND GPLv2+ AND MIT
-Vendor:         Microsoft Corporation
-Distribution:   Mariner
-Group:          System Environment/Security
-URL:            https://www.freedesktop.org/wiki/Software/systemd/
-Source0:        https://github.com/%{name}/%{name}-stable/archive/refs/tags/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
-Source1:        95-disable-systemd-oomd.preset
-BuildRequires:  audit-devel
-BuildRequires:  cryptsetup-devel
-BuildRequires:  docbook-dtd-xml
-BuildRequires:  docbook-style-xsl
-BuildRequires:  gettext
-BuildRequires:  glib-devel
-BuildRequires:  gperf
-BuildRequires:  intltool
-BuildRequires:  kbd
-BuildRequires:  kmod-devel
+Url:            https://systemd.io
+%if %{without inplace}
+Version:        255
+%else
+# determine the build information from local checkout
+Version:        %(tools/meson-vcs-tag.sh . error | sed -r 's/-([0-9])/.^\1/; s/-g/_g/')
+%endif
+Release:        2%{?dist}
+
+# FIXME - hardcode to 'stable' for now as that's what we have in our blobstore
+%global stable 1
+#global stable %(c="%version"; [ "$c" = "${c#*.*}" ]; echo $?)
+
+# For a breakdown of the licensing, see README
+License:        LGPL-2.1-or-later AND MIT AND GPL-2.0-or-later
+Summary:        System and Service Manager
+
+# download tarballs with "spectool -g systemd.spec"
+%if %{defined commit}
+Source0:        https://github.com/systemd/systemd%{?stable:-stable}/archive/%{commit}/%{name}-%{shortcommit}.tar.gz
+%else
+%if 0%{?stable}
+Source0:        https://github.com/systemd/systemd-stable/archive/v%{version_no_tilde}/%{name}-%{version_no_tilde}.tar.gz
+%else
+Source0:        https://github.com/systemd/systemd/archive/v%{version_no_tilde}/%{name}-%{version_no_tilde}.tar.gz
```

Comment:
I don't understand. If we don't use the `commit` macro and `stable` is hard-coded (and I believe we don't really want to use non-stable versions), then why the `if-else`?

Reply:
direct from fedora

Reply:
If it's not used, we try to clean-up the specs of any bits which only make sense for other distros (to the best of our abilities and time constraints), so that it's easier to understand the spec. If we're not using these bits, it would help to remove them.